aepsych/benchmark/problem.py

```diff
@@ -316,22 +316,26 @@ def evaluate(self, strat: Union[Strategy, SequentialStrategy]) -> Dict[str, floa
 class LSEProblemWithEdgeLogging(LSEProblem):
     eps = 0.05
 
-    def __init__(self, thresholds):
+    def __init__(self, thresholds: Union[float, List, torch.Tensor]) -> None:
         super().__init__(thresholds)
 
-    def evaluate(self, strat):
+    def evaluate(self, strat: Union[SequentialStrategy, Strategy]) -> Dict[str, float]:
         metrics = super().evaluate(strat)
 
         # add number of edge samples to the log
 
         # get the trials selected by the final strat only
-        n_opt_trials = strat.strat_list[-1].n_trials
+        if isinstance(strat, SequentialStrategy):
+            n_opt_trials = strat.strat_list[-1].n_trials
+        elif isinstance(strat, Strategy):
```

Comment:
Is there ever a case where we pass in a Strategy instead of a SequentialStrategy?

Reply:
You're right, but this is only achieved by removing the `Strategy` type hint from all `Problem` classes.